internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -836,6 +842,35 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	return resourceRead(ctx, d, meta)
 }
 
+func isUpdateAllowed(d *schema.ResourceData) (bool, error) {
+	cs, us := d.GetChange("replication_specs")
+	currentSpecs, updatedSpecs := cs.([]any), us.([]any)
+
+	isNewSchemaCompatible := checkNewSchemaCompatibility(currentSpecs)
+
+	for _, specRaw := range updatedSpecs {
+		if specMap, ok := specRaw.(map[string]any); ok && specMap != nil {
+			numShards, _ := specMap["num_shards"].(int)
+			if numShards > 1 && isNewSchemaCompatible {
+				return false, fmt.Errorf("cannot increase num_shards to >1 under the current configuration. Please modify your resource as outlined in our migration guide before updating to add additional shards")
```

Comment:
```suggestion
				return false, fmt.Errorf("Cannot increase num_shards to > 1 under the current configuration. New shards can be defined by adding new replication spec objects; please refer to our examples, documentation, and 1.18.0 migration guide for more details.")
```

Reply:
Slight rephrasing being more direct on action that would need to be taken if the user wants to add additional shards.